dev-support/git-hooks/pre-commit/100-verify-submodules-pushed.sh

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Redirect output to stderr.
+exec 1>&2
+
+
+#set -o xtrace
+set -o errexit
+set -o pipefail
+set -o nounset
+
+bin="$(cd "$(dirname "$0")" > /dev/null; pwd)"
+
+# Status Table
+# A         Added
+# C         Copied
+# D         Deleted
+# M         Modified
+# R         Renamed
+# T         Type Changed (i.e. regular file, symlink, submodule, …<200b>)
+# U         Unmerged
+# X         Unknown
+# B         Broken
```

Comment:
dead code for now, I don't have a good reason to use, but I do like having this documented in a easy manner to check quickly.